### PR TITLE
New case of image_create_with_large_size

### PR DESCRIPTION
**Pull request by hellohellenmao**
Test creating/resizing an image with large size.

ID: 1797850

Signed-off-by: Tingting Mao <timao@redhat.com>

### CODE REVIEW

File: qemu/tests/cfg/image_create_with_large_size.cfg [hunk 1]
```diff
@@ -0,0 +1,12 @@
+- image_creation_lock_release:
+    virt_test_type = qemu
+    type = image_create_with_large_size
+    start_vm = no
+    create_image = yes
```
Comment (lolyu) [inline qemu/tests/cfg/image_create_with_large_size.cfg]: why `create_image` is `yes`?

File: qemu/tests/cfg/image_create_with_large_size.cfg [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- image_creation_lock_release:
+    virt_test_type = qemu
+    type = image_create_with_large_size
```
Comment (lolyu) [inline qemu/tests/cfg/image_create_with_large_size.cfg]: `only qcow2`?

File: qemu/tests/cfg/image_create_with_large_size.cfg [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    start_vm = no
+    create_image = yes
+    remove_image = yes
+    images = "large middle"
```
Comment (lolyu) [inline qemu/tests/cfg/image_create_with_large_size.cfg]: `middle` is a misleading variable name.

File: qemu/tests/cfg/image_create_with_large_size.cfg [hunk 4]
```diff
@@ -0,0 +1,12 @@
+- image_creation_lock_release:
```
Comment (lolyu) [inline qemu/tests/cfg/image_create_with_large_size.cfg]: `image_creation_lock_release`?

File: qemu/tests/cfg/image_create_with_large_size.cfg [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    virt_test_type = qemu
+    type = image_create_with_large_size
+    start_vm = no
+    create_image = yes
+    remove_image = yes
+    images = "large middle"
+    image_name_large = "images/large"
+    image_name_middle = "images/middle"
```
Comment (lolyu) [inline qemu/tests/cfg/image_create_with_large_size.cfg]: missing `image_format = qcow2`?

File: qemu/tests/image_create_with_large_size.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+from virttest import data_dir
+from virttest.qemu_storage import QemuImg
+
+
+def run(test, params, env):
+    """
+    Creating image with large size.
+    1. Create a qcow2 image with large size and check the output info.
+    2. Create a qcow2 with a normal size.
+    3. Increase a large size to the qcow2 image and check the output info.
```
Comment (lolyu) [inline qemu/tests/image_create_with_large_size.py]: an blank line here would be nicer.

File: qemu/tests/image_create_with_large_size.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    :param test: Qemu test object.
```
Comment (lolyu) [inline qemu/tests/image_create_with_large_size.py]: `test` is actually a vt test object

File: qemu/tests/image_create_with_large_size.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    large_img = params["images"].splite()[1]
+    middle_img = params["images"].splite()[-1]
```
Comment (lolyu) [inline qemu/tests/image_create_with_large_size.py]:
```python
large_img, small_img = params["images"].split()
```

File: qemu/tests/image_create_with_large_size.py [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    root_dir = data_dir.get_data_dir()
+    large = QemuImg(params.object_params(large_img), root_dir, large_img)
+    middle = QemuImg(params.object_params(middle_img), root_dir, middle_img)
+    large_filename = large.image_filename
+    size_increases = params["size_changes"]
+    create_err_info = "The image size is too large for file format " \
+                      "'qcow2' (try using a larger cluster size)"
+    resize_err_info = "Failed to grow the L1 table: File too large"
+
```
Comment (lolyu) [inline qemu/tests/image_create_with_large_size.py]: better to define a function to wrap the `try` block.

File: qemu/tests/image_create_with_large_size.py [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    except exceptions.TestError as err:
+        if create_err_info not in str(err) or large_filename not in str(err):
+            test.fail("CML failed with unexpected output: %s" % err)
+    else:
+        test.fail("There is no error when creating an image with large size.")
+
+    logging.info("Test resizing an image with large size.")
+    small.create(small.params)
+    result = small.resize(size_increases)
+    status, output = result.exit_status, result.stdout.decode()
```
Comment (lolyu) [inline qemu/tests/image_create_with_large_size.py]: `result.stdout_text`

File: qemu/tests/image_create_with_large_size.py [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    else:
+        test.fail("There is no error when creating an image with large size.")
+
+    logging.info("Test resizing an image with large size.")
+    small.create(small.params)
+    result = small.resize(size_increases)
+    status, output = result.exit_status, result.stderr.decode()
+    if status == 0:
+        test.fail("There is no error when resizing an image with large size.")
+    else:
```
Comment (lolyu) [inline qemu/tests/image_create_with_large_size.py]: `elif`

File: qemu/tests/image_create_with_large_size.py (hunk 9 above)
Comment (hellohellenmao) [reply qemu/tests/image_create_with_large_size.py]: For `resize` function change to assert the output not `exception`, there seems no need to define a function. Check again, please.

File: qemu/tests/image_create_with_large_size.py (hunk 11 above)
Comment (hellohellenmao) [reply qemu/tests/image_create_with_large_size.py]: Yes, thanks.

**Comment by hellohellenmao** [comment]
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.image_create_with_large_size: PASS (4.03 s)

**Comment by hellohellenmao** [comment]
@YongxueHong Help to take a review, please.

File: qemu/tests/image_create_with_large_size.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    """
+    large_img, small_img = params["images"].split()
+    root_dir = data_dir.get_data_dir()
+    large = QemuImg(params.object_params(large_img), root_dir, large_img)
+    small = QemuImg(params.object_params(small_img), root_dir, small_img)
+    large_filename = large.image_filename
+    size_increases = params["size_changes"]
+    create_err_info = "The image size is too large for file format " \
+                      "'qcow2' (try using a larger cluster size)"
+    resize_err_info = "Failed to grow the L1 table: File too large"
```
Comment (YongxueHong) [inline qemu/tests/image_create_with_large_size.py]: Suggest that move the `create_err_info` and `resize_err_info` to cfg file to avoid hard code.
Thanks.

Comment (hellohellenmao) [reply qemu/tests/image_create_with_large_size.py]: Updated, thanks.

**Comment by hellohellenmao** [comment]
(1/1) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.image_create_with_large_size: PASS (4.98 s)

**Comment by hellohellenmao** [comment]
@lolyu @YongxueHong Help to review again, please.

**Pull request merged**